packages/editor/src/components/provider/index.js

```diff
@@ -51,7 +51,22 @@ class EditorProvider extends Component {
 
 	getBlockEditorSettings( settings, meta, onMetaChange, reusableBlocks ) {
 		return {
-			...settings,
+			...pick( settings, [
```

Comment:
I kinda feel this should occur in the reducer, both (a) to avoid leaning so much on memoization with derived settings objects (they shouldn't be changing so much) and (b) since it's still technically possible to pass invalid/unrecognized settings when I render my own block editor, and thus the original concern remains.

Reply:
Interesting idea and yeah even if it's a denormalized value, I think it should be fine in the reducer. I'm going to merge now as I'd like to include this in the release (it fixes a bug #14141)